README.md

```diff
@@ -136,12 +136,38 @@ It has two key functions:
    - `p` *must* have been returned by a previous call to `allocate(bytes)`, otherwise behavior is undefined
 
 It is up to a derived class to provide implementations of these functions. 
-See [available resources](#availble-resources) for example `device_memory_resource` derived classes. 
+See [available resources](#available-resources) for example `device_memory_resource` derived classes. 
 
 Unlike `std::pmr::memory_resource`, `rmm::mr::device_memory_resource` does not allow specifying an alignment argument. 
 All allocations are required to be aligned to at least 256B. 
 Furthermore, `device_memory_resource` adds an additional `cudaStream_t` argument to allow specifying the stream on which to perform the (de)allocation.
 
+### Stream-ordered Memory Allocation
+
+`rmm::mr::device_memory_resource` is a base class that provides stream-ordered memory allocation.
+This allows optimizations such as re-using memory deallocated on the same stream without the overhead
+of synchronization.
+
+A call to `device_memory_resource::allocate(bytes, stream_a)` returns a pointer that is valid to use
+on `stream_a`. Using the memory on a different stream (say `stream_b`) is Undefined Behavior unless
+the two streams are first synchronized, for example by using `cudaStreamSynchronize(stream_a)` or by
+recording a CUDA event on `stream_a` and then calling `cudaStreamWaitEvent(stream_b, event)`.
+
+The stream specified to `device_memory_resource::deallocate` should be a stream on which it is valid
+to use the deallocated memory immediately for another allocation. Typically this is the stream
+on which the allocation was *last* used before the call to `deallocate`. The passed stream may be
+used internally by a memory_resource for managing available memory with minimal synchronization, and
```

Comment:
Markdown gets silly about underscores sometimes - recommending code font here. Also, does `memory_resource` mean `device_memory_resource`? If so, it might be clearer to use the full class name. It appears that all memory resources using aforementioned streams (device memory resources) inherit from `device_memory_resource`.
```suggestion
used internally by a `memory_resource` for managing available memory with minimal synchronization, and
```

Reply:
Fixed, thanks.